main.go

```diff
@@ -41,10 +41,15 @@ func main() {
 	printVersionInfo()
 
 	mgrConfig := controllers.ManagerConfig{}
-	controllers.InitFromFlags(flag.CommandLine, &mgrConfig)
+	if err := controllers.InitFromFlags(flag.CommandLine, &mgrConfig); err != nil {
+		logger.Error(err, "failed to initialize from flags")
+		os.Exit(1)
+	}
 
 	flag.Parse()
-	printFlags(logger)
+
+	flags := getFlags()
+	printFlags(logger, flags)
```

Comment:
This change is no longer necessary, since we have no use case now for storing `flags` and using it anywhere else other than in `printFlags`. You can revert this to the previous state where flags were fetched and printed within `printFlags` function.